File: clase12-01-2019/Programa clase.py
import cv2
import numpy as np

lower_blue = np.array([98,99,87])
upper_blue = np.array([115,255,255])

lower_pink = np.array([160,50,50])
upper_pink = np.array([180,255,255])

lower_green = np.array([50,100,100])
upper_green = np.array([70,255,255])

lower_red = np.array([0,100,100])
upper_red = np.array([10,255,255])

lower_white = np.array([0,0,100])
upper_white = np.array([0,0,255])

lower_black = np.array([0,0,0])
upper_black = np.array([255,255,55])

cap= cv2.VideoCapture(0)
while(True):
    ret, frame = cap.read()
    gray= cv2.cvtColor(frame, cv2.COLOR_BGR2GRAY)
    #cv2.imshow('gray',gray)
    blur= cv2.GaussianBlur(frame,(7,7),0)
    #cv2.imshow('blur',blur)
    hsv= cv2.cvtColor(frame, cv2.COLOR_BGR2HSV)
    #cv2.imshow('hsv',hsv)
    union= cv2.inRange(hsv, lower_blue, upper_blue)
    #cv2.imshow('blue',union)
    f1= cv2.erode(union, cv2.getStructuringElement(cv2.MORPH_RECT,(3,3)), iterations=1)
    f2= cv2.erode(f1, cv2.getStructuringElement(cv2.MORPH_RECT,(5,5)), iterations=1)
    object= cv2.moments(f2)
    if object['m00'] > 50000:
        cx= int(object['m10']/object['m00'])
        cy= int(object['m01']/object['m00'])
        cv2.circle(frame, (cx,cy), 10, (0,255,0), 2)
    cv2.imshow('rgb',frame)
    if cv2.waitKey(1) & 0xFF == ord('q'):
        break

cap.release()
cv2.destroyAllWindows()

File: clase12-01-2019/programas clase/programaClase.cpp

//g++ $(pkg-config --cflags --libs opencv) -std=c++11  programaClase.cpp -o yourFileProgram
#include "opencv2/opencv.hpp"
using namespace cv;
int main(int, char**)
{
    Mat gray;
    Mat hsv;
    Mat mask;
    Mat filtro1, filtro2;
    double dM01, dM10, dArea;
    VideoCapture cap(0); 
    if(!cap.isOpened()) 
        return -1;

    for(;;)
    {
        Mat frame;
        cap >> frame; 
        cvtColor(frame, gray, COLOR_BGR2GRAY);
        imshow("Gray", gray);
        cvtColor(frame, hsv, COLOR_BGR2HSV);
        imshow("HSV", hsv);
        Scalar minBlue = cv::Scalar(98,99,87); 
        Scalar maxBlue = cv::Scalar(115,255,255); 
        inRange(hsv, minBlue, maxBlue, mask);
        imshow("blue", mask);
        erode(mask, filtro1, getStructuringElement(MORPH_RECT, Size(3, 3)) );
        erode(filtro1, filtro2, getStructuringElement(MORPH_RECT, Size(5, 5)) );

        Moments oMoments = moments(filtro2);
        dM01 = oMoments.m01;
        dM10 = oMoments.m10;
        dArea = oMoments.m00;
        if (dArea > 50000)
        {
            int posX = dM10 / dArea;
            int posY = dM01 / dArea;
            circle(frame, Point(posX,posY), 10, Scalar(0,255,0), 2);
        }
        imshow("frame", frame);
        if(waitKey(30) >= 0) break;
    }
    
    return 0;
}